src/mono/mono/mini/method-to-ir.c

```diff
@@ -7457,7 +7458,18 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 			HandleCallData cdata;
 			memset (&cdata, 0, sizeof (HandleCallData));
 
-			cmethod = mini_get_method (cfg, method, token, NULL, generic_context);
+			// The method to be called may have already been resolved when handling a previous opcode. In that
+			// case, we ignore the operand and act as CALL, instead of CALLVIRT.
+			// E.g. #32166 (box+callvirt optimization)
```

Comment:
nit: using the full issue link is preferable (we've been through a couple issue tracking systems so better to be explicit)
```suggestion
			// E.g. https://github.com/dotnet/runtime/issues/32166 (box+callvirt optimization)
```